### ISSUE TITLE
[BUG] - Can't distribute app when adding Adyen using SPM

### ISSUE THREAD

**Issue by JAIRMG** [Adyen/adyen-ios#1526]
**Describe the bug**
We started migrating from cocoapods to SPM and we are facing an issue when we integrate adyen [5.6.0](https://github.com/Adyen/adyen-ios/releases/tag/5.6.0) using SPM

**To Reproduce**
1. Import Adyen using SPM
2. Archive  (this step is succesful)
3. Distribute: Here is failing with error:  

***No architectures in the binary. Lipo failed to detect any architectures in the bundle executable***

**Screenshots**
![Screenshot 2024-02-21 at 15 50 03](https://github.com/Adyen/adyen-ios/assets/26467027/236cac4a-6128-4d5a-9d33-c45f57105e1b)
![Screenshot 2024-02-21 at 15 50 34](https://github.com/Adyen/adyen-ios/assets/26467027/cae4e253-0471-4f47-b0f8-48c4357c94b4)

**Environment**

 - Adyen API Version: 5.6.0
 - Package management system: SPM
 - Xcode version: XCode 15 and 15.2 have same behavior
 - Mac OS type: Intel and M1 same behavior

**Relevant source code**
I tried adding the post-action script as suggested [here](Swift Package Manager for Xcode 12.0 and 12.1 has a [know issue](https://bugs.swift.org/browse/SR-13343) when it comes to importing a dependency that in turn depend on a binary dependencies. A workaround is described [here](https://forums.swift.org/t/swiftpm-binarytarget-dependency-and-code-signing/38953) my scheme without success

**Additional context**
With cocoapods we don't have problems

**Comment by goergisn** [comment]
Hi @JAIRMG,

I just tried to reproduce the issue by building a small example app (incl. uploading to appstore connect) but can't reproduce the issue you are describing.
Is there a way for you to isolate the issue or provide some more implementation details that could lead to this behavior?

Best,
Alex

**Comment by JAIRMG** [comment]
Hi @goergisn thanks for the quick response!

 I found the problem. It was on my side. I had a run script that I added before to remove unused architectures and it was having problems with Adyen3DS2. I found the log on my CI and it says:

Adyen3DS2.framework/Adyen3DS2) must be a fat file when the -extract option is specified
INFO [2024-02-20 14:16:52.96]: ▸ ❌ fatal error: lipo: can't open input file: .../Adyen3DS2.framework/Adyen3DS2-arm64 (No such file or directory)

So after removing that script my upload was successful. Thanks for your help!
